src/main/scala/com/fulcrumgenomics/util/Sequences.scala

```diff
@@ -55,25 +58,73 @@ object Sequences {
     * @return a tuple of (0-based index, length) of the longest homopolymer in the string
     */
+  @deprecated("Use maxPolyX instead", since="0.7.0")
   def longestHomopolymer(s: String): (Int, Int) = {
-    val xs = s.toUpperCase
-    val len = xs.length
-    assert(len > 0, "Cannot compute longest homopolymer of a zero length string")
+    assert(s.length > 0, "Cannot compute longest homopolymer of a zero length string")
+    val result = maxPolyX(s)
+    (result.offset, result.length)
+  }
+
+
+  case class OffsetAndLength(offset: Int, length: Int)
+
+  /**
+    * Returns the offset (0-based) and length of the longest homopolymer in the sequence. In the case
+    * that there are multiple homopolymers of the same length, the earliest one is returned.
+    *
+    * @param s a DNA or RNA sequence
+    * @return the offset and length of the longest homopolymer
+    */
+  def maxPolyX(s: String) : OffsetAndLength = {
+    val bs = s.getBytes
+    var (bestStart, bestLength) = (0,0)
+    var start = 0
+    while (start < bs.length) {
+      val firstBase = s.charAt(start).toByte
+      var i = start + 1
+      while (i < s.length && SequenceUtil.basesEqual(firstBase, bs(i))) i += 1
+      val length = i - start
+      if (length > bestLength) {
+        bestStart = start
+        bestLength = length
+      }
+      start = i
+    }
 
-    var foundIndex = 0
-    var foundLength = 1
+    OffsetAndLength(bestStart, bestLength)
+  }
 
-    forloop (from=0, until=len) { i =>
-      val x = xs.charAt(i)
-      var j = i+1
-      while (j < len && xs.charAt(j) == x) j += 1
-      val newLength = j - i
-      if (newLength > foundLength) {
-        foundIndex = i
-        foundLength = newLength
+  /**
+    * Returns the offset (0-based) and length of the longest dinucleotide run in the sequence. In the case
+    * that there are multiple dinucleotide runs of the same length, the earliest one is returned.
+    *
+    * @param s a DNA or RNA sequence
+    * @return the offset and length of the longest dinucleotide sequence
+    */
+  def maxDinuc(s: String) : OffsetAndLength = {
+    val bs = s.toUpperCase.getBytes
+    var (bestStart, bestLength) = (0,0)
+    forloop(0, bs.length-1) { start =>
+      val (b1, b2) = (bs(start), bs(start+1))
+      var i = start
+      while (i < s.length-1 && SequenceUtil.basesEqual(b1, bs(i)) && SequenceUtil.basesEqual(b2, bs(i+1))) i += 2
+      val length = i - start
+      if (length > bestLength) {
+        bestStart = start
+        bestLength = length
       }
     }
 
-    (foundIndex, foundLength)
+    OffsetAndLength(bestStart, bestLength)
+  }
+
+  /** Reverse complements a string of bases. */
+  def reverseComplement(s: String): String = SequenceUtil.reverseComplement(s)
+
+  /** Returns the sequence that is the complement of the provided sequence. */
+  def complement(s: String): String = {
+    val bs = s.getBytes
+    for (i <- 0 until bs.length) bs(i) = SequenceUtil.complement(bs(i))
```

Comment:
I think `forloop(from=0, until=bs.length)` will be much faster here.

Reply:
It is, fixed.